cpp/dolfinx/mesh/Mesh.cpp

```diff
@@ -322,8 +322,10 @@ mesh::create_submesh(const Mesh& mesh, int dim,
   // NOTE: Unclear what this return for prisms
   const std::size_t num_entity_dofs = layout.num_entity_closure_dofs(dim);
 
-  std::vector<std::int32_t> geometry_indices(num_entity_dofs
-                                             * submesh_to_mesh_map.size());
+  std::vector<std::int32_t> geometry_indices;
+  geometry_indices.reserve(num_entity_dofs * submesh_to_mesh_map.size());
+  std::vector<std::int32_t> submesh_x_dofmap_offsets(1, 0);
+  submesh_x_dofmap_offsets.reserve(submesh_to_mesh_map.size() + 1);
```

Comment:
I would raise this up one line, and follow it by `submesh_x_dofmap_offsets.push_back(0)`.